### DISCUSSION TITLE
Log error records as a pipe delimited string

### DISCUSSION THREAD

**Discussion by dinakarmu** [oracle/python-oracledb#388] [Q&A] [answered]
I have a list of tuples that will be loaded using executemany. Good records should get loaded while I want to log the error or bad records into an error_table with three columns run_date, data_string, error. is this possible?

**Comment by anthony-tuininga** [comment, marked as answer]
Yes, it is definitely possible! Use the `batcherrors` option for `executemany()` which is documented [here](https://python-oracledb.readthedocs.io/en/latest/user_guide/batch_statement.html#handling-data-errors).

**Reply by dinakarmu** [reply]
Should I be loading the error records in a loop one by one from the "data" list in the example? I can't use executemany() to load the error records in one shot?

**Reply by anthony-tuininga** [reply]
You can use executemany() to load the error records as well. No reason not to!
